coffea/processor/executor.py

```diff
@@ -109,10 +115,15 @@ def chunks(self, target_chunksize, align_clusters):
         else:
             n = max(round(self.metadata["numentries"] / target_chunksize), 1)
             actual_chunksize = math.ceil(self.metadata["numentries"] / n)
-            for index in range(n):
-                start, stop = actual_chunksize * index, min(
-                    self.metadata["numentries"], actual_chunksize * (index + 1)
-                )
+            next_chunksize = None
+
+            start = 0
+            while start < self.metadata["numentries"]:
+                if dynamic_chunksize:
+                    next_chunksize = yield
```

Comment:
I think this `yield` is extraneous: if you instead capture the `next_chunksize` on the line that yields the WorkItem, there is only one yield per loop iteration. On the other hand, you need to prime this function with an initial chunksize so that `next_chunksize` is initialized.